### ISSUE TITLE
[BUG/ISSUE] Running Hg simulation with inter-annually varying anthropogenic emissions

### ISSUE THREAD

**Issue by arifein** [geoschem/geos-chem#1195]
### What institution are you from?
MIT

### Description of the problem
I am trying to run an Hg simulation with inter-annually varying anthropogenic emissions (e.g. 2014–2015 with varying emissions according to the Streets et al. 2019 inventory). If I submit a run for 2 years:

`Start YYYYMMDD, hhmmss  : 20140101 000000
End   YYYYMMDD, hhmmss  : 20151201 000000`

The second year of anthropogenic emissions is not updated, with emissions from 2015-01 beng identical to 2014–01. See attached figures for output from `GEOSChem.MercuryEmis` diagnostics, with the `EmisHg0anthro` showing exactly the same range in 2014 and 2015. If instead I run 2014 and then restart the simulation in 2015, `EmisHg0anthro` shows the correct 2015 limits. Biomass burning emissions of Hg0 (`EmisHg0biomass`), which also vary inter-annually, do not show the same issue and correctly retrieve the 2015 emissions. 

### Description of troubleshooting performed
I tried this in GEOS-Chem v12.8.1 and 13.3.4. The HEMCO line is:
`0 STREETS_HG0 $ROOT/MERCURY/v2021-09/Streets/Streets2019_Hg.nc Hg0 2000-2015/1/1/0 C xy kg/m2/s Hg0 - 1 1
`
I also tried to split up the file into the different years explicitly:
`0 STREETS_HG0 /net/fs03/d0/arifein/emissions/v2021-09/Streets2019_Hg_$YYYY.nc Hg0 2000-2015/1/1/0 C xy kg/m2/s Hg0 - 1 1
`
I also tried with different HEMCO CRE codes (R, RF). I also see that the file is being read in the output log, but somehow it is not passed on to the Hg code:
```
> DATE: 2015/01/01  UTC: 00:00  X-HRS:   8760.000000
HEMCO: Opening /net/fs03/d0/arifein/emissions/v2021-09/Streets2019_Hg_2015.nc
```

Has anyone else run an Hg simulation with inter-annually varying anthropogenic emissions? 

### GEOS-Chem version
12.8.1, 13.3.4

### Ncview Plots
**Correct 2014 emissions**
<img width="479" alt="2014_emis" src="https://user-images.githubusercontent.com/82537515/160426338-67ee52e2-cec1-4254-8d2f-8ced08be3d59.png">

**Wrong 2015 emissions (repeated 2014 when running 2 years)**
<img width="474" alt="2015_emis_wrong" src="https://user-images.githubusercontent.com/82537515/160426365-0be4795b-cbd6-4186-b183-5e98ade218b5.png">

**Correct 2015 emissions (restarting simulation in 2015-01)**
![2015_emis_correct](https://user-images.githubusercontent.com/82537515/160426901-f16e1b36-95c2-4aba-8f4b-96291a7b3981.png)

**Comment by yantosca** [comment]
Thanks for writing @arifein.  I am not sure the EmisMercury collection was ever validated.  In the new code that @viral211 sent us for inclusion into 13.4.0, we now save Hg diagnostics to the HEMCO_diagnostics file, which should be more accurate.

**Comment by CharikleiaGRN** [comment]
Same for me. I checked both log files and output from some simulations with updated Streets. There is no inter-annual variability.

**Comment by arifein** [comment]
I tested now the EDGAR inventory in v13.3.3 and it shows the same behaviour. So it does seem like an issue with the diagnostics rather than the inventory, but I'm not sure if this also means that the incorrect amounts are emitted into the model.

**Comment by yantosca** [comment]
Also tagging @viral211.

**Comment by yantosca** [comment]
Would someone be able to take a look into this: @jennyfisher @cdholmes @yanxuz ?

**Comment by viral211** [comment]
I think this might be happening because we read in the anthropogenic Hg emissions only at the first time step.
https://github.com/geoschem/geos-chem/blob/9ab6931db5b556eafd0e002e9fd4fdbcbd5fbb44/GeosCore/mercury_mod.F90#L2423-L2438

We don't do this for biomass burning Hg emissions, which are calculated at each emissions time step.

We could fix this by reading in Hg anthropogenic emissions at the beginning of each year, by replacing 
`IF ( FIRST ) THEN` in Line 2424 to `IF (  ITS_A_NEW_YEAR() ) THEN`

**Comment by yantosca** [comment]
Thanks @viral211!  That should fix it!

In 13.4.0, I believe this code has been removed, so that all Hg emissions come through the HEMCO state and not through the manual diagnostics.

**Comment by arifein** [comment]
Thanks a lot @viral211 ! I tested out this fix, with also including an extra use statement for the subroutine `EMISSMERCURY`  on line 2348: `USE TIME_MOD,           ONLY : GET_MONTH, ITS_A_NEW_MONTH, ITS_A_NEW_YEAR`

I ran a 13 month simulation (2014-2015) in 4x5 resolution in GCv13.3.4. I now get slightly different anthropogenic emissions for 2015-01 (max emissions Hg0 = 0.00161301 kg/s) compared to 2014-01 (max emissions = 0.00161274 kg/s), but still different then if I would restart directly in 2015-01 (max emissions Hg0 = 0.00164559). So it seems like there may still be an issue with this diagnostic and emissions. The code change may also have affected other diagnostics, with ~2x increases in the EmisHg0snow max and 10x increases in FluxHg2HgPfromAirToSnow.

I am glad that this will be improved upon in v13.4! I will try to keep looking in to what the issue may be, but happy if others have feedback.

**Comment by viral211** [comment]
Thanks for checking on this @arifein. Another possible workaround is to just remove the `IF` statement on line 2424. I commented out lines 2424 and 2436-2438 and ran a simulation for 2014-12-31-2015-01-02 and got the right emissions. Could you try this?

**Comment by arifein** [comment]
Hi @viral211, just tried this and it works! Thanks for solving that

**Comment by yanxuz** [comment]
Hi Ari @arifein, my student Peng Zhang run the simulation version 13.3.1 a while ago and found some issues in HgII vertical distribution and wet deposition to ocean. He also told me some of the emission diagnostics have issues as well (EmisHg0geogenic, EmisHg0biomass, EmisHg0anthro are all zeros). Please see this file for more detail: 
[GC13.3.1-test.docx](https://github.com/geoschem/geos-chem/files/8456974/GC13.3.1-test.docx)

Not quite sure if things are corrected in 13.3.4. Could you have a look at it? Thanks.

**Comment by yantosca** [comment]
Thanks all for looking into this.  When 13.4.0 ships, we will need the Hg and POPs user community to help validate the scientific output of the model (as this is beyond the scope of what the GCST can do).  13.4.0 has been delayed a bit due to some bugs caught in the benchmarking process, so we will get this out as soon as we can.
